src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react'
+import { RouteComponentProps } from 'react-router-dom'
+import { KeyManager, SUPPORTED_WALLETS } from 'context/WalletProvider/config'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import { LocationState } from '../../NativeWallet/types'
+import { ActionTypes, useWallet, WalletActions } from '../../WalletProvider'
+
+export interface MetaMaskSetupProps
+  extends RouteComponentProps<
+    {},
+    any, // history
+    LocationState
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
+  const { dispatch, state } = useWallet()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  // eslint-disable-next-line no-sequences
+  const setErrorLoading = (e: string | null) => (setError(e), setLoading(false))
```

Comment:
what's going on here? does this just call both functions and return void?

Reply:
That's correct. This is just a helper function that sets loading to false whenever the error message is set.